flow360/component/simulation/cloud.py

```diff
@@ -183,6 +190,18 @@ def _run(
             f"params argument must be a SimulationParams object but is of type {type(params)}"
         )
 
+    ##-- Getting the project length unit from draft and store in the SimulationParams
+    _resp = RestApi(ProjectInterface.endpoint, id=source_asset.project_id).get()
+    last_opened_draft_id = _resp["lastOpenDraftId"]
+    assert last_opened_draft_id is not None
+    _draft_with_length_unit = Draft.from_cloud(last_opened_draft_id)
+    length_unit = _draft_with_length_unit.retrieve_simulation_dict()[
+        "private_attribute_asset_cache"
+    ]["project_length_unit"]
```

Comment:
I guess we need some error handling here

Reply:
How user provides this value for the first time, before any project or draft is created?

Reply:
When calling `from_file` method user has to specify the length unit. Then it passed to the project API.

Reply:
Right. Raised key error if the length unit is not found.